beacon_node/execution_layer/src/lib.rs

```diff
@@ -441,72 +443,80 @@ impl ExecutionLayer {
     /// failure) from all nodes and then return based on the first of these conditions which
     /// returns true:
     ///
+    /// - Error::ConsensusFailure if some nodes return valid and some return invalid
     /// - Valid, if any nodes return valid.
     /// - Invalid, if any nodes return invalid.
     /// - Syncing, if any nodes return syncing.
     /// - An error, if all nodes return an error.
-    pub async fn execute_payload<T: EthSpec>(
+    pub async fn notify_new_payload<T: EthSpec>(
         &self,
         execution_payload: &ExecutionPayload<T>,
-    ) -> Result<(ExecutePayloadResponseStatus, Option<Hash256>), Error> {
+    ) -> Result<(PayloadStatusV1Status, Option<Vec<Hash256>>), Error> {
         debug!(
             self.log(),
-            "Issuing engine_executePayload";
+            "Issuing engine_newPayload";
             "parent_hash" => ?execution_payload.parent_hash,
             "block_hash" => ?execution_payload.block_hash,
             "block_number" => execution_payload.block_number,
         );
 
         let broadcast_results = self
             .engines()
-            .broadcast(|engine| engine.api.execute_payload_v1(execution_payload.clone()))
+            .broadcast(|engine| engine.api.new_payload_v1(execution_payload.clone()))
             .await;
 
         let mut errors = vec![];
         let mut valid = 0;
         let mut invalid = 0;
         let mut syncing = 0;
-        let mut invalid_latest_valid_hash = vec![];
+        let mut invalid_latest_valid_hash = HashSet::new();
         for result in broadcast_results {
-            match result.map(|response| (response.latest_valid_hash, response.status)) {
-                Ok((Some(latest_hash), ExecutePayloadResponseStatus::Valid)) => {
-                    if latest_hash == execution_payload.block_hash {
-                        valid += 1;
-                    } else {
+            match result {
+                Ok(response) => match (&response.latest_valid_hash, &response.status) {
+                    (Some(latest_hash), &PayloadStatusV1Status::Valid) => {
+                        if latest_hash == &execution_payload.block_hash {
+                            valid += 1;
+                        } else {
+                            errors.push(EngineError::Api {
+                                id: "unknown".to_string(),
+                                error: engine_api::Error::BadResponse(
+                                    format!(
+                                        "new_payload: response.status = Valid but invalid latest_valid_hash. Expected({:?}) Found({:?})",
+                                        execution_payload.block_hash,
+                                        latest_hash,
+                                    )
+                                ),
+                            });
+                            // I've commented out the lines below because this is a malformed response
```

Comment:
I agree with you on this one. 

The [spec](https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#specification) says the `latestValidHash` should always be `null`, however I'm open to accepting the `execution_payload.block_hash` value in the interests of being liberal with what we accept (whilst still being correct).

So, my suggestion is to remove the commented-out lines :slightly_smiling_face: